### DISCUSSION TITLE
External Datasource Display Sum in Design

### DISCUSSION THREAD

**Discussion by hendrikjap** [Budibase/budibase#4190] [Help] [answered]
Hi Guys,

In Data, I created a column in formula as below
![image](https://user-images.githubusercontent.com/55825618/150965080-862addba-8ac7-4928-9158-7e5d4699f30e.png)

In Design, I want to sum all purchase items and display it 
![image](https://user-images.githubusercontent.com/55825618/150965169-226c7fcc-b577-4d80-980f-8b11a7347c8a.png)

However, it seems it doesn't add the number, instead it is concatenated them
![image](https://user-images.githubusercontent.com/55825618/150965318-23d53946-a94d-4b6b-a5d9-aa4c7e737001.png)

This is the JS formula I used in the design
![image](https://user-images.githubusercontent.com/55825618/150965524-b9e03b18-0e61-4880-a6df-8c96ea0ebc76.png)

Anything is wrong?

**Comment by mjashanks** [comment, marked as answer]
I see what you mean - Budibase is treating `amount` as if it were `text`, instead of a `number`.

A workaround is to wrap with `parseFloat`...

```javascript
sum = sum + parseFloat(purchaseitems.amount);
```

**Reply by hendrikjap** [reply]
Ahhh,, it works. Thanks!

**Comment by aptkingston** [comment]
Hey @ely2727. This has been fixed in #4086 and will be included in our next release 👍
